### PR TITLE
Subject example service integration test

### PR DESCRIPTION
**Pull request by vldmrklnk**

### CODE REVIEW

File: Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            var addNewSubjectResponse = await comprefaceApiClient.SubjectService.AddSubject(new AddSubjectRequest() { Subject = TEST_SUBJECT_NAME });
+            var subjectExample = new AddSubjectExampleRequest()
+            {
+                DetProbThreShold = 0.81m,
+                FilePath = FILE_PATH,
+                Subject = addNewSubjectResponse.Subject,
+                FileName = FILE_NAME
+            };
+
+            var expectedAddExampleSubjectResponse = await comprefaceApiClient.SubjectExampleService.AddSubjectExampleAsync(subjectExample);
```
Comment (Evgeniy-Sinicin) [inline Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs:45]: I think, If you're testing the AddSubjectExampleAsync method, what it returns is the actual value (not expected)... 
In this case, you've tested the GetAllSubjectExampleAsync method 😅
Here either the logic is wrong, or the name of the test method))))
But I don't know exactly... Maybe I'm just not good at testing 🤪

File: Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+            //Arrange
+            var addNewSubjectResponse = await comprefaceApiClient.SubjectService.AddSubject(new AddSubjectRequest() { Subject = TEST_SUBJECT_NAME });
+            var addBase64ExampleSubjectRequest = new AddBase64SubjectExampleRequest()
+            {
+                DetProbThreShold = 0.81m,
+                Subject = addNewSubjectResponse.Subject,
+                File = IMAGE_BASE64_STRING
+            };
+
+            var expectedAddBase64SubjectExampleResponse = await comprefaceApiClient.SubjectExampleService.AddBase64SubjectExampleAsync(addBase64ExampleSubjectRequest);
```
Comment (Evgeniy-Sinicin) [inline Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs]: ... the same story 👆🙃

File: Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+            var addNewSubjectResponse = await comprefaceApiClient.SubjectService.AddSubject(new AddSubjectRequest() { Subject = TEST_SUBJECT_NAME });
+            var allSubjectExamples = new ListAllSubjectExamplesRequest()
+            {
+                Page = 0,
+                Size = 0,
+                Subject = addNewSubjectResponse.Subject
+            };
+
+            var actualListAllExampleSubjectResponse = await comprefaceApiClient.SubjectExampleService.GetAllSubjectExamplesAsync(allSubjectExamples);
+            var actualCount = actualListAllExampleSubjectResponse.Faces.Count;
```
Comment (Evgeniy-Sinicin) [inline Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs]: The variable is never used 🙈

And you're confusing ... 🙃
You expect the method (that you are testing) will return that count of elements
So it's "expected"

File: Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                    {
+                        DetProbThreShold = 0.81m,
+                        Subject = addNewSubjectResponse.Subject,
+                        File = IMAGE_BASE64_STRING
+                    });
+            }
+
+            //Act
+            var expectedAllSubjectExamplesResponse = await comprefaceApiClient.SubjectExampleService.GetAllSubjectExamplesAsync(allSubjectExamples);
+            var expectedCount = expectedAllSubjectExamplesResponse.Faces.Count;
```
Comment (Evgeniy-Sinicin) [inline Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs]: ... and this is the real count of items returned by the method
So it's "actual"

File: Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+﻿namespace Exadel.Compreface.AcceptenceTests
+{
+    public static class UrlConstConfig
```
Comment (Evgeniy-Sinicin) [inline Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs:3]: This place deprived me of sleep 😄🙈

File: Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs [hunk 6]
```diff
@@ -0,0 +1,254 @@
+﻿using Exadel.Compreface.Configuration;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.AddBase64ExampleSubject;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.AddExampleSubject;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DeleteAllSubjectExamples;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DeleteImageById;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DeleteMultipleExamples;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DownloadImageById;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.DownloadImageBySubjectId;
+using Exadel.Compreface.DTOs.ExampleSubjectDTOs.ListAllExampleSubject;
+using Exadel.Compreface.DTOs.HelperDTOs;
```
Comment (Ukrainiangirl) [inline Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs]: Where do you use it? Exadel.Compreface.DTOs.HelperDTOs

File: Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+using Exadel.Compreface.DTOs.SubjectDTOs.AddSubject;
+using Exadel.Compreface.DTOs.SubjectDTOs.DeleteSubject;
+using Exadel.Compreface.Exceptions;
```
Comment (Ukrainiangirl) [inline Exadel.Compreface.AcceptenceTests/Services/SubjectExampleServiceTest.cs]: the same Exadel.Compreface.Exceptions;

File: Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+        public const string API_KEY_VERIFICATION_SERVICE = "d2403a77-ae38-41af-ab94-8fb82515dd59";
+
+        public const string SUBJECT_NAME = "Stars";
+
+        public const string TEST_SUBJECT_NAME = "Test Subject Name";
+
+        public const string FILE_NAME = "brad-pitt_24.jpg";
+
+        public const string FILE_PATH = "C:\\Users\\ukalenik\\Desktop\\brad-pitt_24.jpg";
```
Comment (Ukrainiangirl) [inline Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs]: move this file to project

File: Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+        //public const string FACE_DETECTION_SERVICE_QUERY = "detection/detect/";
+
+        //public const string EXAMPLE_SUBJECT_SERVICE_QUERY = "recognition/faces/";
+
+        //public const string RECOGNITION_SERVICE_QUERY = "recognition/recognize";
+
+        //public const string SUBJECT_SERVICE_QUERY = "recognition/recognize";
+
+        public const string API_KEY_DETECTION_SERVICE = "d2403a77-ae38-41af-ab94-8fb82515dd59";
```
Comment (Ukrainiangirl) [inline Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs]: add description. Where do we take this Guid

File: Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs (hunk 8 above)
Comment (vldmrklnk) [reply Exadel.Compreface.AcceptenceTests/UrlConstConfig.cs]: Done

**Review by Evgeniy-Sinicin — DISMISSED** [review]

**Review by Evgeniy-Sinicin — APPROVED** [review]
Everything seems to be fine 👍

**Review by asindarov — APPROVED** [review]

**Pull request merged**